src/main/java/org/motechproject/mots/web/UserController.java

```diff
@@ -86,6 +89,35 @@ public UserDto saveUser(@PathVariable("id") UUID id, @RequestBody UserDto userDt
     return userMapper.toDto(userService.saveUser(existingUser));
   }
 
+  /**
+   * Update User password.
+   * @param currentPassword user previous password, should be the same as current password
+   * @param newPassword user's new password
+   * @return updated User
+   */
+  @RequestMapping(value = "/user/password/change/{currentPassword}/{newPassword}",
+      method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.OK)
+  @ResponseBody
+  public ResponseEntity<String> changeCurrentUserPassword(
+      @PathVariable("currentPassword") String currentPassword,
+      @PathVariable("newPassword") String newPassword) {
+
+    String userName =
+        (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+    User existingUser = userService.getUserByUserName(userName);
+
+    if (existingUser == null
+        || !PasswordValidator.validateCurrentPassword(
+            existingUser.getPassword(), currentPassword)) {
+      return new ResponseEntity<>("Current password is incorrect.", HttpStatus.BAD_REQUEST);
+    }
+
+    userService.changeUserPassword(existingUser, newPassword);
+
+    return new ResponseEntity<>("Password changed successfully.", HttpStatus.OK);
```

Comment:
we're not using the response entity in MOTS, return a String, you already set a status with annotation